numpy/random/_common.pyx

```diff
@@ -232,13 +232,35 @@ cdef validate_output_shape(iter_shape, np.ndarray output):
         )
 
 
-cdef check_output(object out, object dtype, object size):
+cdef check_output(object out, object dtype, object size, bint require_c_array):
+    """
+    Check user-supplied output array properties and shape
+    
+    Parameters
+    ----------
+    out : {ndarray, None}
+        The array to check.  If None, returns immediately.
+    dtype : dtype
+        The required dtype of out.
+    size : {None, int, tuple[int]}
+        The size passed.  If out is an ndarray, verifies that the shape of out
+        matches size.
+    require_c_array : bool
+        Whether out must be a C-array.  If False, out can be either C- or F-
+        ordered.  If True, must be C-ordered. In either case, must be
+        contiguous, writable and aligned.
+    """
     if out is None:
         return
     cdef np.ndarray out_array = <np.ndarray>out
     if not (np.PyArray_CHKFLAGS(out_array, np.NPY_CARRAY) or
-            np.PyArray_CHKFLAGS(out_array, np.NPY_FARRAY)):
-        raise ValueError('Supplied output array is not contiguous, writable or aligned.')
+            (np.PyArray_CHKFLAGS(out_array, np.NPY_FARRAY) and
```

Comment:
Note that `NPY_CARRAY` and `NPY_FARRAY` were deprecated in 1.7. The new names are `NPY_ARRAY_CARRAY` and `NPY_ARRAY_FARRAY` . You could also  use `PyArray_ISCARRAY(...)` and `PyArray_ISFARRAY(...)` instead.

Reply:
Replaced with the function versions.